fpdf/image_parsing.py

```diff
@@ -127,24 +132,160 @@ def get_img_info(img, image_filter="AUTO", dims=None):
         ):
             info["smask"] = _to_data(img, image_filter, select_slice=alpha_channel)
 
+    dp = f"/Predictor 15 /Colors {dpn} /Columns {w}"
+    
+    if img.mode == "1":
+        inverted = False
+        # copied from img2pdf
+        if (
+            img.format == "TIFF"
+            and img.info["compression"] == "group4"
+            and len(img.tag_v2[TiffImagePlugin.STRIPOFFSETS]) == 1
+            and len(img.tag_v2[TiffImagePlugin.STRIPBYTECOUNTS]) == 1
+        ):
+            photo = img.tag_v2[TiffImagePlugin.PHOTOMETRIC_INTERPRETATION]
+            if photo == 0:
+                inverted = True
+        if inverted:
+            dp = f"/BlackIs1 true /Columns {w} /K -1 /Rows {h}"
+        else:
+            dp = f"/BlackIs1 false /Columns {w} /K -1 /Rows {h}"
+
     info.update(
         {
             "w": w,
             "h": h,
             "cs": colspace,
             "bpc": bpc,
             "f": image_filter,
-            "dp": f"/Predictor 15 /Colors {dpn} /Columns {w}",
+            "dp": dp,
         }
     )
 
     return info
 
+class temp_attr:
+    """
+    Copied verbatim from img2pdf
```

Comment:
Note that using from `img2pdf` is OK, as `fpdf2` uses the same license: [LGPL v3](https://gitlab.mister-muffin.de/josch/img2pdf/src/branch/main/LICENSE). Otherwise it might not be allowed.

No need to mention the copying in the docstring though.

Reply:
I checked the license yes, it was quite a relief! I'll remove the mention from the docstrict